compiler/rustc_mir_transform/src/shim.rs

```diff
@@ -90,7 +90,11 @@ fn make_shim<'tcx>(tcx: TyCtxt<'tcx>, instance: ty::InstanceDef<'tcx>) -> Body<'
     };
     debug!("make_shim({:?}) = untransformed {:?}", instance, result);
 
-    pm::run_passes(
+    // We don't validate MIR here because the shims may generate code that's
+    // only valid in a reveal-all param-env. However, since we do initial
+    // validation with the MirBuilt phase, which uses a user-facing param-env.
```

Comment:
That probably works. I'm not sure if that breaks any other invariants of validation, let me try it.

Reply:
This does not work, because validation fails before the `Derefer` is run.

Reply:
I went down this path last week but wasn't very happy with the result as I had to run `Derefer` and then it seemed wise to run the other passes to actually put us in the same state as we should be when `phase = Runtime(Initial)` but that required either duplicating the list of passes or splitting up `run_runtime_lowering_passes()` since running `ElaborateDrops` again caused other issues. In the end, it felt pretty hacky and I think @compiler-errors's solution here is better.